src/Symmetric.jl

```diff
@@ -0,0 +1,213 @@
+@doc doc"""
+    SymmetricMatrices{N,T} <: Manifold
```

Comment:
I prefer the name `Symmetric` for consistency with what we already have for `SymmetricPositiveDefinite`. Are there any downsides to renaming it?

Reply:
I chose SymmetricMatrices (instead of Symmetric), because I think the name is not as ambiguous, as "Symmetric" could also mean symmetric functions (or anything symmetric).

Reply:
I would also opt for SymmetricMatrices, since `Symmetric` is the command from `LinearAlgeba` that symmetrises a matrix.

Reply:
Oh of course, I didn't think of that. I completely agree.